### PR TITLE
AR Example App

### PR DESCRIPTION
**Pull request by tonyskansf**
This PR contains a WIP example app for the AR lecture.

The application's task is to show a bounding box helping users measure the necessary dimensions for a packaging box.

** Features **
- Show the bounding box on tap gesture. 
  - _Note: The initial requirement was to show the bounding box around a detected real-world object; however, this task is firstly not always accurate and would need much more knowledge, which I think would be way out of the lecture's scope._
- Move the bounding box around with a pan gesture.
- Scale the bounding box with a pinch gesture.
- Resize respective sides/faces of the bounding box by dragging one of its faces with a long press gesture.

### CODE REVIEW

File: DimensionsAR/DimensionsAR/Extensions/simd_float3+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//  DimensionsAR
+//
+//  Created by Tony Ngo on 05.10.2022.
+//
+
+import SceneKit
+
+extension simd_float3 {
+    /// Creates a new vector from the simd_float4 — omitting the fourth element of the vector.
+    init(_ simd: simd_float4) {
```
Comment (cejanen) [inline DimensionsAR/DimensionsAR/Extensions/simd_float3+Extensions.swift:15]: well, can we add some description what is the fourth parameter we erase?

File: DimensionsAR/DimensionsAR/Application/ARViewController.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+extension ARViewController {
+    func setup() {
+        gestureManager.setupGestures(in: sceneView)
+        setupSceneView()
+        setupDimensionsView()
+    }
+
+    func setupSceneView() {
+        // Set the view's delegate
+        sceneView.delegate = self
```
Comment (cejanen) [inline DimensionsAR/DimensionsAR/Application/ARViewController.swift:116]: we don't use any delegate methods yet, is it necessary to have this set? or will we add something?

File: DimensionsAR/DimensionsAR/Extensions/ARSCNView+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+//  ARSCNView+Extensions.swift
+//  DimensionsAR
+//
+//  Created by Tony Ngo on 05.10.2022.
+//
+
+import ARKit
+
+extension ARSCNView {
+    /// Returns the projection of a point from 2D view onto a plane in its coordinate system.
```
Comment (cejanen) [inline DimensionsAR/DimensionsAR/Extensions/ARSCNView+Extensions.swift:11]: what is plane?

**Review by cejanen — APPROVED** [review]
Nice...
I approve just few notes. 

I have also some questions to figure out either in code or in presentation. Also I have to say I was a bit confused by ARWorld prefixes :-D and geometry naming.
1) Bounding box works independently with 3 dimensions even though it's cube. Is that ok?
2) I have issue to draw box tapping on screen. What should I be looking for? A raycast from sceneView in gestureManager return empty results.
3) There are many things to be more described in the code (maybe it is in presentation) but those we will add at the end to do a good mix with presentation.
4) box is not very visible I would just do it less transparent
5) the bottom panel on the camera with frames etc is by default? what are those values? should we hide it?

File: DimensionsAR/DimensionsAR/Application/ARViewController.swift (hunk 2 above)
Comment (tonyskansf) [reply DimensionsAR/DimensionsAR/Application/ARViewController.swift:116]: I left it there as I was experiencing with some collision magic. I might add some occlusion so I'll leave it for now.

File: DimensionsAR/DimensionsAR/Extensions/ARSCNView+Extensions.swift (hunk 3 above)
Comment (tonyskansf) [reply DimensionsAR/DimensionsAR/Extensions/ARSCNView+Extensions.swift:11]: Refers to a flat surface, not to be confused with a plane ✈️. 😅 We can say this as a note during the presentation, but I didn't find it necessary to mention it here as its part of the graphics terminology.

Is that ok, or do you think we should explain it here as well? @cejanen

**Comment by tonyskansf** [comment]
> 2) I have issue to draw box tapping on screen. What should I be looking for? A raycast from sceneView in gestureManager return empty results.

I think sometimes under bad conditions the hit test fails. I'll try to explicitly add plane detection and see if it helps.

> 3) There are many things to be more described in the code (maybe it is in presentation) but those we will add at the end to do a good mix with presentation.

I've gone through the code again and have added some extra comments. Let me know if there are any other parts that needs more describing.

> 5) the bottom panel on the camera with frames etc is by default? what are those values? should we hide it?

It is not. It set with `sceneView.showsStatistics = true`. If needed we can hide it.

**Pull request merged**